camel/toolkits/hybrid_browser_toolkit/ws_wrapper.py

```
@@ -0,0 +1,487 @@
+        # Connect to the WebSocket server
+        try:
+            self.websocket = await websockets.connect(
+                f"ws://localhost:{self.server_port}",
+                ping_interval=30,
+                ping_timeout=10,
+                max_size=50 * 1024 * 1024,  # 50MB limit to match server
+            )
+            logger.info("Connected to WebSocket server")
+        except Exception as e:
+            self.process.kill()
+            raise RuntimeError(f"Failed to connect to WebSocket server: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)